lightning/src/routing/gossip.rs

```diff
@@ -130,6 +134,18 @@ pub struct NetworkGraph<L: Deref> where L::Target: Logger {
 	// Lock order: channels -> nodes
 	channels: RwLock<BTreeMap<u64, ChannelInfo>>,
 	nodes: RwLock<BTreeMap<NodeId, NodeInfo>>,
+	#[cfg(feature = "std")]
+	/// Keeps track of [`NodeId`]s we have explicitly removed due to permanent failure so that we
+	/// don't resync them from gossip. Each [`NodeId`] is mapped to the instant when we removed it
+	/// from the network graph so that we can forget we removed it after some time (by removing it
+	/// from this map) and it can be resynced from gossip if it appears again.
+	removed_nodes: Mutex<HashMap<NodeId, Instant>>,
```

Comment:
Are we sure it belongs here, and not at one level above (GossipSync)?

Reply:
Makes sense. In either case, I wanted to keep it close to the network graph. Not opposed to moving it.

Reply:
Perhaps @TheBlueMatt has a better idea. Maybe we just don't care, in the end it's not like it's Bitcoin Core. In that sense, having it inside `NetworkGraph` saves us some code I think.